sql/parser/eval.go

```diff
@@ -771,40 +1166,26 @@ func evalRangeCond(expr *RangeCond) (Datum, error) {
 	// AND left <= to". The only tricky part is that we evaluate "left" only
 	// once.
 
-	left, err := EvalExpr(expr.Left)
-	if err != nil {
-		return DNull, err
-	}
-
-	limits := [2]struct {
-		op   ComparisonOp
-		expr Expr
-	}{
-		{GE, expr.From},
-		{LE, expr.To},
+	var newExpr Expr = &AndExpr{
```

Comment:
It was a simplification that I stumbled upon when I was writing the typeCheck equivalent of this function. Given that NormalizeExpr does this, I'll just remove it altogether.